sql/moz-fx-data-shared-prod/firefox_desktop_derived/urlbar_events_v2/query.sql

```diff
@@ -0,0 +1,119 @@
+ CASE
+    WHEN event_name = 'engagement'
+      AND (
+        (
+          engagement_type IN (
+            "click",
+            "drop_go",
+            "enter",
+            "go_button",
+            "paste_go"
+          )
+        )
+      )
+      THEN 'engaged'
+    WHEN event_name = 'abandonment'
+      THEN 'abandoned'
+    WHEN event_name = 'engagement'
+      AND (
+        (
+          engagement_type NOT IN (
+            "click",
+            "drop_go",
+            "enter",
+            "go_button",
+            "paste_go"
+          )
+        )
+      )
+      THEN "annoyance"
+    ELSE NULL
+  END
+);
+
+CREATE TEMP FUNCTION get_is_terminal(selected_result STRING, engagement_type STRING) AS (
+  --events where the urlbar dropdown menu remains open (i.e., the urlbar session did not end)
+  COALESCE(
+    NOT (selected_result = 'tab_to_search' AND engagement_type IN ('click', 'enter', 'go_button'))
+    AND NOT (
+      selected_result = 'tip_dismissal_acknowledgement'
+      AND engagement_type IN ('click', 'enter')
+    )
+    AND NOT (
+      engagement_type IN (
+        'dismiss',
+        'inaccurate_location',
+        'not_interested',
+        'not_relevant',
+        'show_less_frequently'
+      )
+    ),
+    TRUE
+  )
+);
+
+WITH events_unnested AS (
+SELECT
+  DATE(submission_timestamp) AS submission_date,
+  client_info.client_id AS glean_client_id,
+  metrics.uuid.legacy_telemetry_client_id AS legacy_telemetry_client_id,
+  sample_id,
+  name AS event_name,
+  timestamp AS event_timestamp,
+  GENERATE_UUID() AS event_id,
+  ping_info.experiments,
+  ping_info.seq,
+  normalized_channel,
+  normalized_country_code,
+  `moz-fx-data-shared-prod`.udf.normalize_search_engine(mozfun.map.get_key(extra, "search_engine_default_id")) AS normalized_engine,
+  COALESCE(metrics.boolean.urlbar_pref_suggest_data_collection, FALSE) AS pref_data_collection,
+  COALESCE(metrics.boolean.urlbar_pref_suggest_sponsored, FALSE) AS pref_sponsored_suggestions,
+  COALESCE(metrics.boolean.urlbar_pref_suggest_nonsponsored, FALSE) AS pref_fx_suggestions,
+  mozfun.map.get_key(extra, "engagement_type") AS engagement_type,
+  mozfun.map.get_key(extra, "interaction") AS interaction,
+  CAST(mozfun.map.get_key(extra, "n_chars") AS int) AS num_chars_typed,
+  CAST(mozfun.map.get_key(extra, "n_results") AS int) AS num_total_results,
+  --If 0, then no result was selected.
+  NULLIF(CAST(mozfun.map.get_key(extra, "selected_position") AS int), 0) AS selected_position,
+  mozfun.map.get_key(extra, "selected_result") AS selected_result,
+  enumerated_array(SPLIT(mozfun.map.get_key(extra, "results"), ','), SPLIT(mozfun.map.get_key(extra, "groups"), ',')) AS results,
+  FROM
+    `moz-fx-data-shared-prod.firefox_desktop_stable.events_v1`,
+    UNNEST(events)
+  WHERE
+    DATE(submission_timestamp) = @submission_date
+    AND category = 'urlbar'
+    AND name IN ('engagement', 'abandonment')
```

Comment:
Can you add an alias to events? `category` and `name` are both coming from the event, right?
```suggestion
    UNNEST(events) AS event
  WHERE
    DATE(submission_timestamp) = @submission_date
    AND event.category = 'urlbar'
    AND event.name IN ('engagement', 'abandonment')
```